### ISSUE TITLE
DeviantArt not scraping the best available version

### ISSUE THREAD

**Issue by faucetlol2** [Earlopain/reverser#86]
The image provided in the API response seems to be some sort of sample instead of the full size image, this seems to be a recent change because the scraper was previously downloading the full-size images. I've observed it when scraping a few artists now.

```
      "content": {
        "src": "https://images-wixmp-ed30a86b8c4ca887773594c2.wixmp.com/f/b74bd1be-c13f-4c85-a620-a7710716c325/dfup7mn-cfeb11d4-1666-478a-89c0-7de8df6fb06c.png/v1/fill/w_1280,h_1704,q_80,strp/campfire_by_mayrinart_dfup7mn-fullview.jpg?token=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJzdWIiOiJ1cm46YXBwOjdlMGQxODg5ODIyNjQzNzNhNWYwZDQxNWVhMGQyNmUwIiwiaXNzIjoidXJuOmFwcDo3ZTBkMTg4OTgyMjY0MzczYTVmMGQ0MTVlYTBkMjZlMCIsIm9iaiI6W1t7ImhlaWdodCI6Ijw9MTcwNCIsInBhdGgiOiJcL2ZcL2I3NGJkMWJlLWMxM2YtNGM4NS1hNjIwLWE3NzEwNzE2YzMyNVwvZGZ1cDdtbi1jZmViMTFkNC0xNjY2LTQ3OGEtODljMC03ZGU4ZGY2ZmIwNmMucG5nIiwid2lkdGgiOiI8PTEyODAifV1dLCJhdWQiOlsidXJuOnNlcnZpY2U6aW1hZ2Uub3BlcmF0aW9ucyJdfQ.2q80GPrrV177Cf2xlU7E__BuMDCyc5wtl77w6-ZH3UM",
        "height": 1704,
        "width": 1280,
        "transparency": false,
        "filesize": 8084064
      },
```

When actually going to the page for the image (https://www.deviantart.com/mayrinart/art/958557407) the file is available for download and instead downloads a 3758*5000 image. Might look into this myself later, but not very familiar with DeviantArt's API.

**Comment by faucetlol2** [comment]
I see what's going on here, the API seems to be reporting `"is_downloadable": false` on these images despite them actually being downloadable.

**Comment by Earlopain** [comment]
It's probably the same as this https://github.com/mikf/gallery-dl/issues/3845

TLDR: Client credentials don't work for downloading mature content anymore, I will have to go with proper OAuth instead. That's going to involve some browser automation, not really looking forward to that.

**Comment by Earlopain** [comment]
With this fix it should once again get the best version. Take care to correctly set the apps redirect uri.

**Comment by faucetlol2** [comment]
Looks to be working good! Sorry for the late response but it seemed like all the DA artists I was watching went AWOL after this fix so it took forever to see it in practice.
